src/BaseObject.js

```diff
@@ -0,0 +1,145 @@
+  this.gasGatherers = 1;
+
+  //used to offset the rate of resource gathering
+  //but at some point this should be tied into time rather than arbitrary units
+  this.globalRate = 0.01;
+
+  //this is an object that should hold all the info that the base is going to display
+  //should have a type 'label' which doesnt get updated
+  //and allows us to display information about the base
+  this.information = [{name: 'Resources',
+                       type: 'resource', // type may be unecessary
+                       x: 20 + game.world.width,
+                       y: 100,
+                       anchor: [0.0, 0.0],
+                       context: this,
+                       updater: this.getResources},
+                       {name: 'Gatherers',
+                       type: 'gatherers', 
+                       x: 170 + game.world.width,
+                       y: 100,
+                       anchor: [0.0, 0.0],
+                       context: this,
+                       updater: this.getGatherers},
+                       {name: 'abundance',
+                       type: 'abundance', 
+                       x: 300 + game.world.width,
+                       y: 100,
+                       anchor: [0.0, 0.0],
+                       context: this,
+                       updater: this.getAbundance}];
+
+  //and this is the object that actually displays our text
+  //it might not be a bad idea to move all this stuff into the sectordata object
+  //but for the time being it makes more sense here
+  this.infoDisplay = null; // new TextGroup(this, 0, 0, this.information);
+
+};
+
+BaseObject.prototype = {
+  //updates our resource count every frame
+  // TODO decouple this from frame
+  //    accept a delta time in the function 
+  //      in order to do this we need to first keep track of time globally
+  update: function() {
+    this.metalCount += this.globalRate * this.metalGatherers * this.resources.metal.abundance;
+    this.rockCount += this.globalRate * this.rockGatherers * this.resources.rock.abundance;
+    this.liquidCount += this.globalRate * this.liquidGatherers * this.resources.liquid.abundance;
+    this.woodCount += this.globalRate * this.woodGatherers * this.resources.wood.abundance;
+    this.plantCount += this.globalRate * this.plantGatherers * this.resources.plant.abundance;
+    this.gasCount += this.globalRate * this.gasGatherers * this.resources.gas.abundance;
+  },
+
+  //use getters and setters so that the textgroup is initialized when the base is created
+  //not when the sector is initialized
+  get active() {
+    return this._active;
+  },
+
+  set active(x) {
+    this._active = x;
+    this.infoDisplay = new TextGroup(this, 0, 0, this.information);
+  },
+
+  updateText: function() {
+    //update the information here
+    //use the information to selectively update a textGroup
+    if (this.information.length != this.infoDisplay.children.length) {
+      this.infoDisplay.destroy();
```

Comment:
This doesnt happen every time. This is only for when the length of our information object exceeds the length of the information display object. So it only evaluates to true when we add something to the information object. And the reason I destroy it is to ensure the order is kept when adding and removing items to be displayed.